esy/System.ml

```diff
@@ -1,8 +1,12 @@
-type system =
+type t =
   | Darwin
   | Linux
   | Cygwin
-  | Other
+  | Other of string
+
+let isCygwin =
+  let test = Re.(compile (seq [bos; str "cygwin"])) in
```

Comment:
Got this test from https://en.wikipedia.org/wiki/Uname

Reply:
Ah I did something similiar in my PR here too: https://github.com/bryphe/esy/blob/1de1f2cdde4149d7454efbce102a17d39a134c76/esy/System.ml

Reply:
Thanks, used `System.os_type` too.